<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>868</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>10</y>
      <width>431</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>Action</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QPushButton" name="Enable_servo">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>101</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Enable servo</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Stop_btn">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>200</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Go_home_btn">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>200</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Go home</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Move_btn">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>200</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Move to Target</string>
     </property>
    </widget>
    <widget class="QTabWidget" name="tabWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>361</width>
       <height>111</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>axis1</string>
      </attribute>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Target Position</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>40</y>
         <width>69</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_7">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Disable axis1</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>axis2</string>
      </attribute>
      <widget class="QCheckBox" name="check_dis_axis2">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Target Position</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Disable axis1</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_5">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>40</y>
         <width>69</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>axis3</string>
      </attribute>
      <widget class="QCheckBox" name="check_dis_axis3">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Target Position</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Disable axis1</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_6">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>69</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>axis4</string>
      </attribute>
      <widget class="QCheckBox" name="check_dis_axis4">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Target Position</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_17">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Disable axis1</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_7">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>69</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>axis5</string>
      </attribute>
      <widget class="QCheckBox" name="check_dis_axis5">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_18">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Target Position</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Disable axis1</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_8">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>40</y>
         <width>69</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>axis6</string>
      </attribute>
      <widget class="QCheckBox" name="check_dis_axis6">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Set Position</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_21">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Disable axis1</string>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="doubleSpinBox_9">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>69</width>
         <height>26</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QLabel" name="check_save_torque">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>140</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>save torque data</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>save torque data</string>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox_14">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>150</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </widget>
    <widget class="QLabel" name="check_sve_position">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>140</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>save position data</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>270</y>
      <width>431</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QPlainTextEdit" name="plainTextEdit">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>431</width>
       <height>191</height>
      </rect>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="clear_dis">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>210</y>
       <width>431</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear Display</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>381</width>
      <height>441</height>
     </rect>
    </property>
    <property name="title">
     <string>Data to Calculate PT</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <widget class="QSpinBox" name="Inter_ponits">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>30</y>
       <width>91</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QLabel" name="Lookahead_points">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>70</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Lookahead points</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="Num_axis">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>140</y>
       <width>91</width>
       <height>26</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="Frequency">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>110</y>
       <width>91</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <number>50000</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QLabel" name="delta_max">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>170</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>delta_max</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>110</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Frequency</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="delta_vmax">
     <property name="geometry">
      <rect>
       <x>208</x>
       <y>170</y>
       <width>91</width>
       <height>26</height>
      </rect>
     </property>
     <property name="decimals">
      <number>10</number>
     </property>
     <property name="minimum">
      <double>-99.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
     <property name="value">
      <double>0.000000000000000</double>
     </property>
    </widget>
    <widget class="QSpinBox" name="Lookahead">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>70</y>
       <width>91</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Interpolation points</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>140</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Number of axis</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>310</y>
       <width>311</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Use the different Max Limited Velocity?</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_use_diff_vel">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>320</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
     <property name="tristate">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_use_diff_acc">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>360</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>350</y>
       <width>341</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Use the different Max Limited Acceleration?</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Load_angle_btn">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>400</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Load angle file</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Calculate_pt">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>400</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Calculate PT</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_save_pt">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>400</y>
       <width>21</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="iconSize">
      <size>
       <width>20</width>
       <height>20</height>
      </size>
     </property>
    </widget>
    <widget class="QLabel" name="check">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>390</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Save PT data?</string>
     </property>
    </widget>
    <widget class="QLabel" name="delta_max_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>230</y>
       <width>201</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Default: Max Limited Velocity</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="Max_limited_velocity">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>230</y>
       <width>91</width>
       <height>26</height>
      </rect>
     </property>
     <property name="decimals">
      <number>3</number>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
     <property name="value">
      <double>0.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="Max_limited_acc">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>270</y>
       <width>91</width>
       <height>26</height>
      </rect>
     </property>
     <property name="decimals">
      <number>3</number>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
     <property name="value">
      <double>0.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="delta_max_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>270</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Default: Max LimitedAcceleration</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpProgress">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>440</y>
      <width>391</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>复制进度</string>
    </property>
    <widget class="QProgressBar" name="progressBar">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>391</width>
       <height>30</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QProgressBar" name="progressBar2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>391</width>
       <height>10</height>
      </rect>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="value">
      <number>-1</number>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpOperation1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="title">
     <string>多线程复制（方法：重写run函数）</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="grpOperation2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>570</y>
      <width>231</width>
      <height>31</height>
     </rect>
    </property>
    <property name="title">
     <string>多线程复制（方法：MoveToThread）</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="Cancel_pushButton">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>540</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>暂停计算</string>
    </property>
   </widget>
   <widget class="QPushButton" name="Start_pushButton">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>540</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string> 开始计算</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>868</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
